### PR TITLE
Rocm jaxlib v0.4.30 qa cleanup

### PR DESCRIPTION
**Pull request by hsharsha**

### CODE REVIEW

**Comment by hsharsha** [comment]
@draganmladjenovic  take a look at the failures for MLIR tests at https://github.com/ROCm/xla/pull/35/commits/79b36927dbd3c4fceabb1dcaa4d750ec551ad0ff

Most of them are failing due to bigger thread size and smaller block size in AMD GPUs compared to NVIDIA. Some tests run into infinite loop on MI200 and hence I have commented out RunAndCompare

**Review by i-chaochen — COMMENTED** [review]
I think we're ok on CublasDot at 0.4.28, but it's failed on 0.4.30?

wondering do we have a ticket to track on it?

**Comment by hsharsha** [comment]
> I think we're ok on CublasDot at 0.4.28, but it's failed on 0.4.30?
> 
> wondering do we have a ticket to track on it?

It needs more investigation. It depends on the choice of autotune https://github.com/ROCm/frameworks-internal/issues/9088

File: xla/service/algorithm_util.cc [hunk 1]
```diff
@@ -155,11 +155,17 @@ bool IsSupportedDotAlgorithmOnGpu(
...
+          gpu_compute_capability) &&
+      std::get<se::RocmComputeCapability>(gpu_compute_capability)
+          .gfx9_mi100_or_later();
+
   switch (algorithm) {
     case PrecisionConfig::ALG_DOT_ANY_F8_ANY_F8_F32:
     case PrecisionConfig::ALG_DOT_ANY_F8_ANY_F8_F32_FAST_ACCUM:
       // Other F8 types are actually not supported by NVIDIA GPUs.
-      return is_cuda_ge_ada &&
+      return (is_cuda_ge_ada || is_rocm_mi100_and_above) &&
```
Comment (pemeliya) [inline xla/service/algorithm_util.cc:168]: I am not sure if this is really correct. I guess FP8 support begins from MI300 arch? But I see that we have the same check also on upstream

**Review by pemeliya — APPROVED** [review]
I have reviewed gemm-related changes and buffer_comparator, leaving the remaining LLVM changes to Dragan and Chao

**Comment by pramenku** [comment]
Is this PR merged in QA-31 too? @hsharsha

**Pull request merged**